lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -245,11 +245,31 @@ impl ShardOperation for LocalShard {
 
         let operation_id = {
             let update_sender = self.update_sender.load();
-
             let channel_permit = update_sender.reserve().await?;
 
+            // TODO: We better lock `wal` before `clock_map`, but it does not work, because `wal` use `parking_lot::Mutex`... :/
+
+            if let Some(clock_tag) = &mut operation.clock_tag {
+                let new_tick = clock_tag.clock_tick;
+
+                let old_tick = self
+                    .clock_map
+                    .lock()
+                    .await
+                    .advance_clock_and_correct_tag(clock_tag);
+
+                if new_tick > 0 && new_tick <= old_tick {
+                    return Ok(UpdateResult {
+                        operation_id: None,
+                        status: UpdateStatus::Acknowledged,
+                        clock_tag: Some(*clock_tag),
+                    });
+                }
+            }
```

Comment:
Could we maybe add a comment here, on the outer and inner if-statement, to describe what this does.

I assume the inner if-statement rejects operations with old ticks. But then I find `new_tick <= old_tick` a bit confusing. Would it maybe help to rename `old_tick` to `last_seen_tick` or `known_tick`?

Reply:
Yeah, sure. It's still a WIP, cause I don't like it either and want to come up with a better API/structure for this check.

Reply:
Reworked it, it's still a mess, but so far I'm not sure how to improve it much.

Reply:
Better now. The `let accepted` clarifies a lot.